apps/vmq_server/src/vmq_reg.erl

```diff
@@ -492,32 +501,90 @@ fix_dead_queue(SubscriberId, Subs, {DeadNodes, [Target|Targets], N}) ->
+        _ ->
+            %% Assume all the subscriptions come with the same clean_session flag.
+            [{_, StartClean, _}|_] = Subs,
+            RepairQueueFun =
+                fun() ->
+                        case rpc:call(Target, vmq_reg, replace_dead_queue, [SubscriberId, DeadNodes, StartClean, Subs]) of
+                            ok ->
+                                {DeadNodes, Targets ++ [Target], N + 1};
+                            Error ->
+                                lager:format("repairing dead queue for ~p on ~p failed due to ~p~n", [SubscriberId, Target, Error]),
+                                {DeadNodes, Targets ++ [Target], N}
+                        end
+                end,
+            vmq_reg_sync:sync(SubscriberId, RepairQueueFun, 60000)
+    end.
+
+
+replace_dead_queue(SubscriberId, _DeadNodes, _StartClean = true, _Subs) ->
+    %% At this point we may have a new subscriber on either this or
+    %% other nodes which may have another set of subscriptions than
+    %% what the old node had.
+
+    %% To ensure we end up with a client which has a consistent state,
+    %% we write an empty subscription to the store which will cause
+    %% clients on other nodes to be disconnected (if
+    %% allow_multiple_sessions=false) (TODO: formalize this behaviour
+    %% in a test-case) and they'll then reconnect.
+    Subs = vmq_subscriber:new(true),
+    vmq_subscriber_db:store(SubscriberId, Subs),
+
+    case get_queue_pid(SubscriberId) of
+        not_found ->
+            %% no local queue, so we delete the client information.
+            del_subscriber(SubscriberId),
+            ok;
+        QPid ->
+            %% we force a disconnect to ensure the client reconnects
+            %% and reestablishes consistent metadata.
+            vmq_queue:force_disconnect(QPid, ?ADMINISTRATIVE_ACTION, true),
+            ok
+    end;
+replace_dead_queue(SubscriberId, DeadNodes, _StartClean = false, Subs) ->
+    %% At this point we may have a new subscriber on either this or
+    %% orther nodes and their subscribtions may differ from the one we
+
+    %% here the goal is that there is a (new) queue to receive any
+    %% offline messages until the client reconnects.
+    case vmq_queue_sup_sup:start_queue(SubscriberId) of
+        {ok, true, _OldQPid} ->
+            case vmq_subscriber_db:read(SubscriberId) of
+                undefined ->
+                    %% not clear why we ended up here - better return
+                    %% an error and let the caller retry.
+                    error;
+                LocalSubs ->
+                    NewSubs = rewrite_dead_nodes(LocalSubs, DeadNodes, node(), false),
+                    %% store the updated subs, also to make sure to
+                    %% propagate the new values to all other nodes.
+                    vmq_subscriber_db:store(SubscriberId, NewSubs)
+            end;
+        _ ->
+            %% new queue, just store the updated subscriptions without
```

Comment:
`new` or `existing` queue..?

Reply:
New queue, as above the `{ok, true, _OldQPid}` was handled.